### PR TITLE
Exposing  use_cudnn

### PR DESCRIPTION
**Pull request by chengduoZH**
fix https://github.com/PaddlePaddle/Paddle/issues/7582

What this PR does includes:
1. Exposing use_cudnn to python interface.
2. In Python side, `use_cudnn` is set `True`.
3. In C++ side, `use_cudnn = use_cudnn && is_gpu_place(ctx.GetPlace());`

### CODE REVIEW

**Comment by wangkuiyi** [comment]
@chengduoZH We need an issue to describe the problem to be solved by this PR.

**Comment by chengduoZH** [comment]
@wangkuiyi Thanks!
I have added an issue to describe the problem.

File: python/paddle/v2/fluid/nets.py [hunk 1]
```diff
@@ -38,8 +41,10 @@ def img_conv_group(input,
                    param_attr=None,
                    conv_with_batchnorm=False,
                    conv_batchnorm_drop_rate=None,
+                   conv_use_cudnn=True,
```
Comment (dzhwinter) [inline python/paddle/v2/fluid/nets.py]: `conv_use_cudnn`, `pool_use_cudnn` in argument is not flexible enough.
I think we can expose the 
`HasCUDNN` DSO function,  HasCUDNN will determine use cudnn or not according to the machine environment.

File: python/paddle/v2/fluid/nets.py [hunk 2]
```diff
@@ -38,8 +41,10 @@ def img_conv_group(input,
...
                    pool_stride=1,
-                   pool_type=None):
+                   pool_type=None,
+                   pool_use_cudnn=True):
```
Comment (dzhwinter) [inline python/paddle/v2/fluid/nets.py]: Just use_cudnn, as the last argument.

Comment (chengduoZH) [reply python/paddle/v2/fluid/nets.py]: Done

File: paddle/operators/conv_op.cc [hunk 3]
```diff
@@ -70,6 +70,13 @@ void ConvOp::InferShape(framework::InferShapeContext* ctx) const {
 framework::OpKernelType ConvOp::GetExpectedKernelType(
     const framework::ExecutionContext& ctx) const {
   bool use_cudnn = ctx.Attr<bool>("use_cudnn");
+  use_cudnn &= platform::is_gpu_place(ctx.GetPlace());
```
Comment (jacquesqiao) [inline paddle/operators/conv_op.cc]: I think `use_cudnn` means that user wants to force use cudnn operator, if env does not support cudnn, the framework should throw an error but not implicit change to use other operator.

Comment (dzhwinter) [reply paddle/operators/conv_op.cc]: Agree with the comment.  
We need a default policy right now, by default, we need to set `use_cudnn` True.
As we discussed offline, we reach the agreement that place is also a user configuration. we need to **overwrite our default policy**, not user specified.

Comment (jacquesqiao) [reply paddle/operators/conv_op.cc]: ok, I get the point, seems we still need a priority mechanism to resolve such kind of problem in the future.

**Pull request merged**